cmd/udplogger/main.go

```diff
@@ -42,15 +45,23 @@ func main() {
 		if err != nil {
 			log.Fatal("reading from UDP socket: ", err)
 		}
-		log.Printf("received: %q\n", string(buf[:rlen]))
+		if raw {
+			log.Printf("%v\n", string(buf[:rlen]))
```

Comment:
This adds a newline after the content of each UDP datagram, which can be misleading given the user has explicitly asked for a raw dump of the data received.

I also think the messages from line 20 and 39 should be avoided while in raw mode, for the same reason. That way you can send the output to a file or through a pipe for processing without contaminating the data.

I think it would be sensible that while in raw mode the output should be exactly the contents of the datagrams.
What do you think?

Reply:
I agree about the newline, and I will remove that.

I also understand your point of view about raw actually being **raw**. But I think it should wait for the refactoring, as fixing this now will add a lot more if-statements - that will make the code harder to read. I do not mind the startup-logging....

Reply:
Thanks for addressing the change @erikgb. Sadly, the new version still prints EOL after each datagram content due to how the log package works, see https://golang.org/pkg/log/:

> Every log message is output on a separate line: if the message being printed does not end in a newline, the logger will add one.

But don't worry, I'm going to merge it nonetheless, so you can enjoy the new functionality now. I think it is okay to fix that later since it is still a "secret" functionality (we haven't publish any documentation on how the raw mode works yet). I will fix this and other details in the future when I can spend 1 or 2 hours cleaning this repo a little bit.